src/components/headers_overlay/autoresize_store.ts

```diff
@@ -0,0 +1,89 @@
+import { CellPosition, Command, HeaderIndex, UID } from "../..";
+import { GRID_ICON_MARGIN, ICON_EDGE_LENGTH, PADDING_AUTORESIZE_HORIZONTAL } from "../../constants";
+import {
+  computeIconWidth,
+  computeTextWidth,
+  largeMax,
+  positionToXc,
+  positions,
+  splitTextToWidth,
+} from "../../helpers";
+import { SpreadsheetStore } from "../../stores";
+import { GridCellIconStore } from "../grid_cell_icon_overlay/grid_cell_icon_overlay_store";
+
+export class AutoresizeStore extends SpreadsheetStore {
+  private ctx = document.createElement("canvas").getContext("2d")!;
+
+  protected gridCellIconStore = this.get(GridCellIconStore);
+
+  handle(cmd: Command) {
+    switch (cmd.type) {
+      case "AUTORESIZE_COLUMNS":
+        for (let col of cmd.cols) {
+          const size = this.getColMaxWidth(cmd.sheetId, col);
+          if (size !== 0) {
+            this.model.dispatch("RESIZE_COLUMNS_ROWS", {
+              elements: [col],
+              dimension: "COL",
+              size,
+              sheetId: cmd.sheetId,
+            });
+          }
+        }
+        break;
+      case "AUTORESIZE_ROWS":
+        for (let row of cmd.rows) {
+          this.model.dispatch("RESIZE_COLUMNS_ROWS", {
+            elements: [row],
+            dimension: "ROW",
+            size: null,
```

Comment:
Why don't we compute the size according to the content, as done for the columns ?

Reply:
Row without user defined size (null) are dynamically resized automatically (when there are newlines and stuff). Computing a size here and setting is manually would break this dynamic autoresize.